osquery/tables/system/darwin/xattr.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <string>
+#include <iomanip>
+
+#include <sys/xattr.h>
+
+#include <osquery/logger.h>
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+#include <osquery/core/conversions.h>
+
+namespace osquery {
+namespace tables {
+
+struct XAttrField {
+  uint8_t type;
+  uint8_t header_length;
+  uint64_t length;
+};
+
+std::string handleError(){
+  switch(errno){
+    case ENOATTR:
+      return "No such attribute";
+    case ENOTSUP:
+      return "No system support, or support disabled";
+    case ERANGE:
+      return "Size too small to hold attribute";
+    case EPERM:
+      return "The named attribute is not permitted for this type of object";
+    case EINVAL:
+      return "Name is invalid, or options has an unsupported bit set";
+    case EISDIR:
+      return "The path is not a regular file";
+    case ENOTDIR:
+      return "Part of the path was not a directory";
+    case ENAMETOOLONG:
+      return "The filename was too long";
+    case EACCES:
+      return "Insufficient permissions to read the requested file";
+    case ELOOP:
+      return "Too many symbolic links were encountered";
+    case EFAULT:
+      return "The path or name is invalid";
+    case EIO: 
+      return "There was an I/O error while reading";
+    default:
+      return "No data";
+  }
+}
+
+//Broken out into it's own function so we can use it to get attributes later
+std::string getAttribute(const boost::filesystem::path path, const std::string attribute, int &ret, int &buf_len){
```

Comment:
I would recommend a struct here too, having 3 output variables, none of which indicate error is not very pretty. ;)